src/lib/components/SideNav/SideNav.server.ts

```diff
@@ -0,0 +1,84 @@
+import type { PageMetadataMap, PageMetadataMapItem } from "../../../routes/loadPageMetadataMap";
+import type { NavMenuType } from "$lib/components/Header/Nav/types";
+import type SideNavItem from "./types";
+
+type SideNavMap = Map<string, SideNavItem>;
+
+const sideNavMap: SideNavMap = new Map();
+
+// recursive function that builds out a tree of children, grandchildren, etc.
+const buildSideNavTree = (
+  pageMetadataMap: PageMetadataMap,
+  featuredPages: string[],
+  metadata: PageMetadataMapItem
+): SideNavItem[] => {
+  const tree: SideNavItem[] = [];
+  const { children } = metadata;
```

Comment:
nit: if this is the only thing you need from `metadata`, why not just pass in `metadata.children` as the third argument to `buildSideNavTree`?

Reply:
You're right, good catch! I'll make this change now.

Reply:
Knocked this out in 0ff1908!